File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converters/ConverterListLong.java
package pl.szczodrzynski.edziennik.data.db.converters;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.List;

import androidx.room.TypeConverter;

public class ConverterListLong {

    @TypeConverter
    public static List<Long> toListLong(String value) {
        return value == null ? null : new Gson().fromJson(value, new TypeToken<List<Long>>(){}.getType());
    }

    @TypeConverter
    public static String toString(List<Long> value) {
        return value == null ? null : new Gson().toJson(value);
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converter/ConverterListLong.kt
package pl.szczodrzynski.edziennik.data.db.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class ConverterListLong {
    @TypeConverter
    fun toListLong(value: String?): List<Long>? = value?.let { Gson().fromJson<List<Long>>(it, object : TypeToken<List<Long?>?>() {}.type) }

    @TypeConverter
    fun toString(value: List<Long?>?): String? = value?.let { Gson().toJson(it) }
}
